src/services/codefixes/removeUnnecessaryAwait.ts

```diff
@@ -27,12 +27,34 @@ namespace ts.codefix {
         }
 
         const parenthesizedExpression = tryCast(awaitExpression.parent, isParenthesizedExpression);
-        // (await 0).toFixed() should keep its parens (or add an extra dot for 0..toFixed())
-        if (parenthesizedExpression && isPropertyAccessExpression(parenthesizedExpression.parent) && isDecimalIntegerLiteral(awaitExpression.expression)) {
+        const preserveParens = parenthesizedExpression && (
+            // (await 0).toFixed() should keep its parens (or add an extra dot for 0..toFixed())
+            isPropertyAccessExpression(parenthesizedExpression.parent) && isDecimalIntegerLiteral(awaitExpression.expression, sourceFile) ||
+            // new (await c).Class()
+            isPropertyAccessExpressionInNewExpression(parenthesizedExpression.parent) ||
+            // (await new C).foo
+            isNewExpressionWithoutParens(awaitExpression.expression)
+        );
+
+        if (preserveParens) {
             changeTracker.replaceNode(sourceFile, awaitExpression, awaitExpression.expression);
         }
         else {
             changeTracker.replaceNode(sourceFile, parenthesizedExpression || awaitExpression, awaitExpression.expression);
         }
     }
+
+    function isPropertyAccessExpressionInNewExpression(expression: Node) {
+        return isPropertyAccessExpression(expression) && !!findAncestor(expression, ancestor => {
+            return isPropertyAccessExpression(ancestor)
+                ? false

```

Comment:
unless this is the prettier-generated formatting, I would prefer the lispier

```
return isPropertyAccessExpression(ancestor) ? false
    : isNewExpression(ancestor) ? true
    : "quit";
```

Reply:
Yeah, ternary pyramids are 🤢

Reply:
Looking for lisp? Why not :alien: :squid: :crab: 
```js
cond([[isPropertyAccessExpression(ancestor), false],
      [isNewExpression(ancestor), false],
      [true, "quit"]])
```

Reply:
I was curious; here’s prettier:

```ts
return isPropertyAccessExpression(ancestor)
  ? false
  : isNewExpression(ancestor)
  ? true
  : "quit";
```

🤨

Reply:
That prettier transformation is even harder to read at a glance than the pyramid style, kind of weird.  My favorite is the stacked style @sandersn originally proposed.